tools/CheckMetadata.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python
+
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+import datetime
+import git
+import os
+import re
+import unicodedata
+import unittest
+import urllib
+import yaml
+
+VERSION_PATTERN = r'(\d{4})\.(\d{2})\.(\d{2})(\.\d+)?'
+DOI_PATTERN = r'10.\d{4,9}/zenodo.(\d+)'
```

Comment:
Should the two `.` be escaped here?

Reply:
No, it only needs escaping in a character set like `[\.]`

Reply:
Ohh it's the other way 'round 🤦‍♂️